internal/README.md

```diff
@@ -0,0 +1,116 @@
+# Development Guide: Testing Local test-server Changes
+
+## Overview
+This guide outlines the standard procedure for compiling and testing local modifications to the Go test-server. The test-server is a Go application distributed via language-specific SDK wrappers for ease of adoption.
+
+The recommended workflow for end-to-end validation is to test through the SDK sample. This ensures that any changes to the core Go binary are compatible with the SDK integration layer.
+
+## Step-by-Step e2e Testing Workflow with TypeScript SDK
+
+The process involves compiling your local Go source code, preparing the TypeScript environment, and then linking your new binary to the sample project for testing.
+
+Step 1: Compile the Go Binary
+
+First, compile your local changes into an executable test-server binary.
+
+Navigate to the project's root directory (`test-server`) and build the Go application.
+
+```sh
+go build
+```
+
+After a successful build, a new executable named `test-server` will be present in the root directory.
+
+Step 2: Build the TypeScript SDK Wrapper
+
+The TypeScript SDK acts as a wrapper around the Go binary. It must be built before we can use it in the TypeScript samples.
+
+Navigate to the TypeScript SDK directory: `test-server/sdks/typescript`.
+
+Install dependencies and build the SDK package.
+
+```sh
+npm install
+npm run build
+```
+
+Step 3: Prepare the Sample Test Project
+
+The sample project consumes the TypeScript SDK. We will set it up to run our tests.
+
+Navigate to the sample project directory: `test-server/sdks/typescript/samples`.
+
+Install its dependencies. This command installs the test-server-sdk we just built in step 2.
+
+```sh
+npm install
+```
+
+Step 4: Link Your Local test-server Binary
+This is the most critical step. You must replace the pre-packaged binary in the sample project's dependencies with the custom binary you built in Step 1.
+
+After the previous steps, the directory structure looks like this. The goal is to replace the test-server binary highlighted below, the graph below is a tree view of your directory structure from `test-server/sdks/typescript/samples`.
+
+```diff
```

Comment:
it is nice to add a detailed structure diagram to help understand the structure, my only concern is we may update the structure in the future and may forget to update this structure.

Reply:
The reason want to put it there is, it is really hard to describe which binary to replace, I really want to demonstrate it is the one inside the samples would make it effective. I accidentally replaced the ones that in the sdks/typscript/bin/, didn't work at all. I think the better way is to have the CI run this test can we can refer to CI's workflow for manual test, I will create a internal bug and add the CI and update this documentation. For now, it serve as a starting point for team members to try the local build. :) Hope that is okay.